### PR TITLE
HHH-17912 Disallow multiple valued binding for BasicCollectionType

### PR DESCRIPTION
**Pull request by quaff**
https://hibernate.atlassian.net/browse/HHH-17912

### CODE REVIEW

**Comment by quaff** [comment]
@gavinking  Could you review this PR? thanks.

**Comment by gavinking** [comment]
I mean it looks fine, but I'm not really qualified to say that this is the right way to do it. You need someone who understands that code more intimately.

**Comment by quaff** [comment]
> I mean it looks fine, but I'm not really qualified to say that this is the right way to do it. You need someone who understands that code more intimately.

I guess it's @sebersole the author of `DomainParameterXref .

File: hibernate-core/src/main/java/org/hibernate/query/sqm/internal/DomainParameterXref.java [hunk 1]
```diff
@@ -22,11 +22,13 @@
 import org.hibernate.query.sqm.tree.expression.JpaCriteriaParameter;
 import org.hibernate.query.sqm.tree.expression.SqmJpaCriteriaParameterWrapper;
 import org.hibernate.query.sqm.tree.expression.SqmParameter;
+import org.hibernate.type.BasicCollectionType;
 
 /**
  * Maintains a cross-reference between SqmParameter and QueryParameter references.
  *
  * @author Steve Ebersole
+ * @author Yanming Zhou
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/internal/DomainParameterXref.java:31]: Please don't change author tags for simple changes.  Thanks.

**Comment by sebersole** [comment]
> > I mean it looks fine, but I'm not really qualified to say that this is the right way to do it. You need someone who understands that code more intimately.
> 
> I guess it's @sebersole the author of `DomainParameterXref` .

The change imo is more related with BasicCollectionType, which was added after.

But anyway, the change seems reasonable

File: hibernate-core/src/main/java/org/hibernate/query/sqm/internal/DomainParameterXref.java (hunk 1 above)
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/internal/DomainParameterXref.java:31]:
```suggestion
```

Comment (quaff) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/internal/DomainParameterXref.java]: done.

**Comment by sebersole** [comment]
Pushed to main and 6.5.  Thanks!

**Pull request merged**